quixstreams/sources/community/file/origins/azure.py

```diff
@@ -0,0 +1,51 @@
+from io import BytesIO
+from pathlib import Path
+from typing import Generator, Optional
+
+from .base import FileOrigin
+
+try:
+    from azure.storage.blob import BlobServiceClient
+    from azure.storage.blob._container_client import ContainerClient
+except ImportError as exc:
+    raise ImportError(
+        f"Package {exc.name} is missing: "
+        'run "pip install quixstreams[azure]" to use AzureFileOrigin'
+    ) from exc
+
+__all__ = ("AzureFileOrigin",)
+
+
+class AzureFileOrigin(FileOrigin):
+    def get_folder_count(self, filepath: Path) -> int:
+        data = self.client.list_blobs(name_starts_with=str(filepath))
+        folders = [f for page in data.by_page() for f in page.get("blob_prefixes", [])]
+        return len(folders)
+
+    def __init__(
+        self,
+        connection_string: str,
+        container: str,
+    ):
+        self._client: Optional[ContainerClient] = None
+        self.root_location = container
+        self._credentials = connection_string
+
+    @property
+    def client(self):
+        if not self._client:
+            blob_client = BlobServiceClient.from_connection_string(self._credentials)
+            container_client = blob_client.get_container_client(self.root_location)
+            self._client: ContainerClient = container_client
+        return self._client
+
+    def file_collector(self, folder: Path) -> Generator[str, None, None]:
```

Comment:
Abstract method return type hints at `Iterable[Path]` but this method yields strings.